Simplenote/src/androidTest/java/com/automattic/simplenote/integration/TagDialogFragmentTest.kt

```diff
@@ -0,0 +1,326 @@
+        assertNotNull(tag1)
+        assertNotNull(tag3)
+
+        assertEquals(note1.tags, listOf("tag1", "tag2"))
+        assertEquals(note2.tags, listOf("tag2", "tag3"))
+        assertEquals(note3.tags, listOf("tag1", "tag3"))
+    }
+
+    @Test
+    fun editMergeTagInMultipleNotes() {
+        createTag("tag1")
+        createTag("tag2")
+        createTag("tag3")
+        // To edit tags, tags should belong a note
+        val note1 = createNote("Hello1", listOf("tag1", "tag2"))
+        val note2 = createNote("Hello1", listOf("tag2", "tag3"))
+        val note3 = createNote("Hello1", listOf("tag1", "tag3"))
+
+        assertEquals(tagsBucket.count(), 3)
+        assertEquals(notesBucket.count(), 3)
+
+        launchFragment("tag2")
+
+        onView(allOf(withText("tag2"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        onView(allOf(withText("tag2"), isDescendantOfA(withId(R.id.input_tag_name))))
+                .perform(ViewActions.replaceText("tag1"))
+
+        onView(allOf(withText("tag1"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        val saveText = getResourceString(R.string.save)
+        onView(withText(saveText)).inRoot(isDialog()).check(matches(isDisplayed())).perform(click())
+
+        val canonical = TagUtils.getCanonicalFromLexical(tagsBucket, "tag1")
+        val mergeMessage = getResourceStringWithArgs(R.string.dialog_tag_conflict_message, canonical, "tag2", canonical)
+        onView(withId(R.id.message)).check(matches(withText(mergeMessage)))
+
+        val mergeText = getResourceString(R.string.dialog_tag_conflict_button_positive)
+        onView(withText(mergeText)).perform(click())
+
+        val tag1 = getTag("tag1")
+        val tag2 = getTag("tag2")
+        val tag3 = getTag("tag3")
+
+        assertNull(tag2)
+        assertNotNull(tag1)
+        assertNotNull(tag3)
+
+        assertEquals(note1.tags, listOf("tag1"))
+        assertEquals(note2.tags, listOf("tag3", "tag1"))
+        assertEquals(note3.tags, listOf("tag1", "tag3"))
+    }
+
+    @Test
+    fun editTagFails() {
+        createTag("tag1")
+        // To edit tags, tags should belong a note
+        val note = createNote("Hello World", listOf("tag1"))
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(notesBucket.count(), 1)
+
+        launchFragment("tag1")
+
+        onView(allOf(withText("tag1"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+        onView(allOf(withText("tag1"), isDescendantOfA(withId(R.id.input_tag_name))))
+                .perform(ViewActions.replaceText("tag5"))
+        onView(allOf(withText("tag5"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        tagsBucket.newObjectShouldFail = true
+
+        val saveText = getResourceString(R.string.save)
+        onView(withText(saveText)).inRoot(isDialog()).check(matches(isDisplayed())).perform(click())
+
```

Comment:
Question (❓): In other tests within this test suite you are check whether the actual displayed error message matches the expected one. Can we do that here as well? I am seeing it is a bit complicated to assert that indeed since the error message is an HTML constructed one.

Reply:
Yes, let me check if  I can construct it.

Reply:
I could not find a way to check HTML. I tried with the same message in HTML and it did not work. There is not much information in the Internet either.

Reply:
I agree, it might get tricky indeed. 👌 

What would you say for us to create an issue on it so that to make sure everyone is aware of us knowingly doing so and then this way we can look again in isolation and if time allows later on?

Reply:
Let add the issue and add it in the extra section.

Reply:
Issue: https://github.com/Automattic/simplenote-android/issues/1388